### ISSUE TITLE
Invoice new line price does not automatically update when adding product with predefined price When language is set to Français (Canada)

### ISSUE THREAD

**Issue by philchamp** [Dolibarr/dolibarr#16954]
# Bug
When I try to add a product to an invoice (new or not), the price stays empty. There is nothing wrong happening in the log file set up to debug. By nothing, I mean literally nothing is happening while tailing the log file when I am doing this specific action. I can't remember if it was working in 13.0.0 since I was only building the database before I updated last week to 13.0.2. 

## Environment
- **Version**: [*13.0.2*]
- **OS**: [*Debian 10*]
- **Web server**: [*Apache 2.4.38*]
- **PHP**: [*7.3.19*]
- **Database**: [*MariaDB 5.5.5-10.3.27 *]
- **URL(s)**: [*http://192.168.1.35/dolibarr/compta/facture/card.php?facid=7*]

https://user-images.githubusercontent.com/10824047/112328914-1ffec600-8c8d-11eb-999f-556c1fe2bf2e.mp4

**EDIT**
I guess the issue is either user related or module related because it's working with a lower privileged user who also has less modules activated. It's not working with the admin user. I'll try to some more debugging to see what is responsible for this behaviour.

**EDIT2**
I've enabled all modules on the lower privileged user and it is still working for him. The price on the new product line automatically updates when choosing a product from the list. 

**EDIT 3 AND STEP TO REPRODUCE**
I managed to get it to work for all users. I had to change the language of my users to **English Canada**. My admin language was in French Canada and while the one that was working was in English Canada. I don't know why this is happening while using French instead of English. 

I can reproduce this behaviour if I switch back the language to French Canada.

**Comment by ksar-ksar** [comment]
Hi,

This is working for me.
Seems a JavaScript issue, what is your browser ? Could you try another one ?

**Comment by philchamp** [comment]
> Hi,
> 
> This is working for me.
> Seems a JavaScript issue, what is your browser ? Could you try another one ?

I tried with Chrome and Firefox. I'll try on a different OS. I'm on Arch Linux right now.

**Comment by philchamp** [comment]
> Hi,
> 
> This is working for me.
> Seems a JavaScript issue, what is your browser ? Could you try another one ?

**Comment by philchamp** [comment]
It's not working on Firefox nor Chrome on two separate workstations (Windows 10 and Arch Linux).

**Comment by philchamp** [comment]
bjk

**Comment by Hystepik** [comment]
It seems to be fix so I close the issue.
Feel free to reopen if not.

**Comment by philchamp** [comment]
Not fixed for me. In fact, I got a new instance on a totally different server that has the same issue. I tried chrome and firefox on different desktop and the problem still persists.

Le problème n'est pas réglé pour moi, J'ai même un nouveau déploiement sur un serveur different à un autre établissement qui a le même problème. J'ai essayé Google Chrome et firefox sur plusieurs machines et le problème est toujours présent.

**Comment by eldy** [comment]
Did you try with 13.0.3 ?
You may also have a javascript error when using french canadia. Can you make right click "Inspect" on chrome then tab console.
Then refresh page. Do you see "red line" ? Then select a product. Do yo see red lines ?

**Comment by philchamp** [comment]
@eldy There you go! You were right guys about JavaScript. The error was : `Uncaught SyntaxError: unexpected token: identifier card.php:619:78`
So I inspected this line and it turns out that there is an apostrophe (prix d'entré) in the translation content associated to option value=InputPrice. All I had to do to fix it was overwriting the InputPrice translation directly in Dolibarr by something without an apostrophe (I used "Saisir un prix" like the French translation). The default translation is problematic.
